docs/adr/0002-delivery-channel-creation-and-integration.md

```diff
@@ -0,0 +1,81 @@
+                 |            |
+                 ++--+------+-+
+                  ^  ^      ^
+                  |  |      |
+         +--------+  |      +--------------+
+         |           |                     |
+         v           v                     v
++--------+-+      +--+-------+           +-+--------+
+|          |      |          |           |          |
+| Delivery |      | Delivery |           | Delivery |
+| Channel  |      | Channel  | +-+  +-+  | Channel  |
+| 1        |      | 2        |           | N        |
+|          |      |          |           |          |
+|          |      |          |           |          |
++----------+      +----------+           +----------+
+```
+
+The management api should have knowledge of what delivery channels are available and where certain notification types should be sent. As we send notifications through on a per user per tenant basis, user settings for which communication type(s) a user has allowed is read and a notification is sent to the available delivery channel(s). 
+
+### User notifications and user settings
+
+A user should be able to choose what kind of communication types they receive based on the notification type. There are two ways of approaching this, either:
+1. Send different types of notifications through all channels that are provided by the consumer of this service, and leave the communication type and user configuration for the consumer of the Marain.UserNotifications service. (ie, the consumer of the Marain.UserNotifications service should specify which channels they want to send notifications via an api call when a notification is created)
+2. Store user preferences somewhere that is accessible to the management api, then when a new type of notifications comes through, check what type of communications types  the user has agreed to use and then send a message through the delivery channel .
+
+The latter option was chosen as it provides a user the ability to choose what notifications they want out of the box, if the former is chosen, all users/businesses integrating with this service will then need to write their own notifications user management service.
+
+User settings for which notifications will go to what communication channels for each user will be stored inside a User Notifications Settings table which will be accessed by the Management Api. Configuration in this table will include the Notification Type and the appropriate communication types associated to that notification type. This table will have an associated crud api that will allow a user to view/change the stored settings.
+
+### Delivery Channels and Notification Types
+
+A delivery channel will need two types of setup, the first being the integration of the service with Marain and the second being configuration added in by a user to how this channel will be used in the current implementation (which notification types are permitted to which channels, templating, api keys etc).
```

Comment:
I've removed that paragraph as it didn't make any sense.

I'm not sure exactly how to specify clear terms, where can I declare this?

Reply:
Since it's an area where it's very easy to introduce accidental ambiguity (because there are multiple perspectives on what constitutes a "consumer" "user" or "service") I'd introduce a short section defining terms near the front of the document, like W3C standards usually do.

Reply:
Agreed, will add a section that in.

Reply:
Did this happen? I don't seem to be able to see it.

Reply:
I have added "Consumer" and "User" to a table of definitions at the top of the document.

Reply:
Might be good to pull out the other things too, like "communications type" and "notification" (and anything else that is used more than once to mean something specific!)

Maybe we should consider adding this **Terms** section to our ADR template.